devel/libanalysis/src/module_data_block_vector.c

```diff
@@ -0,0 +1,71 @@
+/*
+   Copyright (C) 2016  Statoil ASA, Norway.
+
+   The file 'module_data_block_vector.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <ert/util/vector.h>
+#include <ert/util/util.h>
+#include <ert/util/type_macros.h>
+
+#include <ert/analysis/module_data_block.h>
+#include <ert/analysis/module_data_block_vector.h>
+
+#define MODULE_DATA_BLOCK_VECTOR_TYPE_ID 732178012
+
+struct module_data_block_vector_struct {
+  UTIL_TYPE_ID_DECLARATION;
+  char                   * ministep_name;
+  vector_type            * data_block_vector;
+};
+
+
+module_data_block_vector_type * module_data_block_vector_alloc( const char* ministep_name) {
+  module_data_block_vector_type * module_data_block_vector = util_malloc( sizeof * module_data_block_vector );
+  module_data_block_vector->ministep_name =util_alloc_string_copy( ministep_name );
+  module_data_block_vector->data_block_vector = vector_alloc_new();
+  return module_data_block_vector;
+}
+
+
+void module_data_block_vector_free( module_data_block_vector_type * module_data_block_vector ) {
+  util_safe_free(module_data_block_vector->ministep_name);
+  vector_free( module_data_block_vector->data_block_vector );
+  free( module_data_block_vector );
+}
+
+void module_data_block_vector_add_data_block( module_data_block_vector_type * module_data_block_vector , const char * datakey) {
```

Comment:
If possible the block structure should be completed in calling scope before it is added, instead of letting this routine create the block.